ingest/src/test/java/com/microsoft/azure/kusto/ingest/E2ETest.java

```diff
@@ -240,6 +238,10 @@ private void assertRowCount(int expectedRowsCount, boolean checkViaJson) {
                     mainTableResult.next();
                     actualRowsCount = mainTableResult.getInt(0) - currentCount;
                 }
+
+
+                Thread.sleep(3000);
+                timeoutInSec -= 3;
```

Comment:
I would move these two lines back to the start of the loop -  this change results in a potential infinite loop when an exception is thrown from the code inside the IF statement.

Reply:
Also, I suggest you replace the while loop with a for loop as you know exactly how many times (max) you want to loop.
This way you don't need to manage timeoutInSec and it helps you skipping the thread.sleep in the first iteration.